libopenage/game_singletons_info.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015-2016 the openage authors. See copying.md for legal info.
+
+#pragma once
+
+#include <string>
+
+#include "gui/guisys/public/gui_singleton_items_info.h"
+
+namespace openage {
+
+class Engine;
+
+namespace gui {
+
+/**
+ * Accessible during creation of any singleton QML item.
+ */
+struct GameSingletonsInfo : GuiSingletonItemsInfo {
+	GameSingletonsInfo(Engine *engine, const std::string &data_dir)
+		:
+		GuiSingletonItemsInfo{},
+		engine{engine},
+		data_dir{data_dir} {
+	}
+
+	Engine *engine;
+	std::string data_dir;
+};
```

Comment:
That's a good idea. This way we can make the engine a regular object, and for the "current" qml and nyan context, those values are directly accessible.